src/link/edit.js

```diff
@@ -102,6 +108,50 @@ const Edit = ( {
 		}
 	};
 
+	const getFormatStart = ( position ) => {
+		if ( value.formats[ position ] && value.formats[ position ][ 0 ].type === formatType ) {
+			return getFormatStart( position - 1 );
+		}
+		return position;
+	};
+
+	const getFormatEnd = ( position ) => {
+		if ( value.formats[ position ] && value.formats[ position ][ 0 ].type === formatType ) {
+			return getFormatEnd( position + 1 );
+		}
+		return position;
+	};
+
+	const getTrimmedStart = ( selectedValue ) => {
+		const selectedString = selectedValue.text.slice( selectedValue.start, selectedValue.end );
+		const trimmed = selectedString.trimStart();
+		if ( selectedString.length !== trimmed.length ) {
+			const delta = selectedString.length - trimmed.length;
+			return selectedValue.start + delta;
+		}
+		return selectedValue.start;
+	};
+
+	const getTrimmedEnd = ( selectedValue ) => {
+		const selectedString = selectedValue.text.slice( selectedValue.start, selectedValue.end );
+		const trimmed = selectedString.trimEnd();
+		if ( selectedString.length !== trimmed.length ) {
+			const delta = selectedString.length - trimmed.length;
+			return selectedValue.end - delta;
+		}
+		return selectedValue.end;
+	};
+
+	const handleTextEdit = () => {
+		const editDetected = lastValue.formats.length !== value.formats.length;
+		const involvesPreviewFormat = lastValue.formats[ value.start ] && lastValue.formats[ value.start ][ 0 ].type === formatType;
+		if ( editDetected && involvesPreviewFormat ) {
+			const formatStart = getFormatStart( value.start - 1 );
+			const formatEnd = getFormatEnd( value.end + 1 );
+			onChange( removeFormat( value, formatType, formatStart, formatEnd ) );
+		}
```

Comment:
> If a user edits a word but then corrects it back to the original word before moving the cursor away, the original preview is restored.

looks like missing the restore feature

Reply:
Yes and as noted elsewhere in another thread, this potentially adds extra complexity: we can try to save some sort of snapshot of the last value when an edit is detected, and then try to compare that with future iterations, but that can get hairy quickly. 

_However_, I noticed that the native undo button works as expected, and I think this is essentially what's desired from the 'restore' feature. Two entry points: `cmd + z` and UI button, please give it a try. I think it's worth discussing this with the rest of the team. 

<img width="326" alt="image" src="https://github.com/wikimedia/wikipediapreview-wordpress/assets/4752599/988d29a4-7706-4ea8-98bd-2beaf2305834">

Reply:
got it 👍